src/main/java/com/glencoesoftware/omero/zarr/OmeroAmazonS3ClientFactory.java

```diff
@@ -55,23 +68,36 @@ protected AWSCredentialsProvider getCredentialsProvider(Properties props) {
                     + " Please use either named profiles or instance"
                     + " profile credentials.");
         }
-        boolean anonymous = Boolean.parseBoolean(
-                (String) props.get("s3fs_anonymous"));
+
+        boolean anonymous = false;
+        try {
+            ProfileCredentialsProvider pcp = new ProfileCredentialsProvider(ANONYMOUS_PROFILE);
+            String keyID = pcp.getCredentials().getAWSAccessKeyId();
+            String secretKey = pcp.getCredentials().getAWSSecretKey();
+            anonymous = ANONYMOUS_PROFILE.equals(keyID) && ANONYMOUS_PROFILE.equals(secretKey);
+
+        } catch (Exception e) {
+            log.error("Failed to create credentials provider for anonymous profile", e);
```

Comment:
Will this create ERROR level statement every time OMERO is connecting to AWS S3 under authentication?

Reply:
this should probably be set to debug